lib/wit/dependency.py

```diff
@@ -68,6 +66,23 @@ def resolve_deps(self, wsroot, repo_paths, download, source_map, packages, queue
 
         return source_map, packages, queue, errors
 
+    def _parallel_clone(self, deps, wsroot, repo_paths, download, errors):
+        def do(dep):
+            try:
+                p = Package(dep.name, repo_paths)
+                p.load(wsroot, download, dep.source, dep.specified_revision)
+            except BadSource as e:
+                errors.append(e)
```

Comment:
I feel like it's pretty unusual in Python to pass in a list as an argument and expect the function to mutably append to it as needed. Could we change this function such that we don't take in the `errors` list as an argument, but we instead build up an errors list and just return it at the end? On the caller side you can still mutate the array, but I think it's best if the caller is responsible for that.

Reply:
I agree that's better, we do do a bit of mutation of arguments in the codebase, but doing it less is ideal